serving/samples/helloworld-kotlin/README.md

```diff
@@ -0,0 +1,245 @@
+        && echo "Adding gradle user and group" \
+        && addgroup -S -g 1000 gradle \
+        && adduser -D -S -G gradle -u 1000 -s /bin/ash gradle \
+        && mkdir /home/gradle/.gradle \
+        && chown -R gradle:gradle /home/gradle \
+        \
+        && echo "Symlinking root Gradle cache to gradle Gradle cache" \
+        && ln -s /home/gradle/.gradle /root/.gradle
+        
+    USER gradle
+    VOLUME "/home/gradle/.gradle"
+    WORKDIR /home/gradle
+
+    ADD build.gradle ./build.gradle
+    ADD src ./src
+    RUN gradle clean build
+    ENTRYPOINT ["java","-jar","-Djava.security.egd=file:/dev/./urandom","/home/gradle/build/libs/gradle.jar"]
+    ```
+
+5. Create a new file, `service.yaml` and copy the following service definition
+   into the file. Make sure to replace `{username}` with your Docker Hub username.
+
+    ```yaml
+    apiVersion: serving.knative.dev/v1alpha1
+    kind: Service
+    metadata:
+      name: helloworld-kotlin
+      namespace: default
+    spec:
+      runLatest:
+        configuration:
+          revisionTemplate:
+            spec:
+              container:
+                image: docker.io/{username}/helloworld-kotlin
+                env:
+                - name: TARGET
+                  value: "Kotlin Sample v1"
+    ```
+
+## Build and deploy this sample
+
+Once you have recreated the sample code files (or used the files in the sample
+folder) you're ready to build and deploy the sample app.
+
+1. Use Docker to build the sample code into a container. To build and push with
+   Docker Hub, run these commands replacing `{username}` with your
+   Docker Hub username:
+
+    ```shell
+    # Build the container on your local machine
+    docker build -t {username}/helloworld-kotlin .
+
+    # Push the container to docker registry
+    docker push {username}/helloworld-kotlin
+    ```
+
+2. After the build has completed and the container is pushed to docker hub, you
+   can deploy the app into your cluster. Ensure that the container image value
+   in `service.yaml` matches the container you built in
+   the previous step. Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f service.yaml
+    ```
+
+3. Now that your service is created, Knative will perform the following steps:
+   * Create a new immutable revision for this version of the app.
+   * Network programming to create a route, ingress, service, and load balance for your app.
+   * Automatically scale your pods up and down (including to zero active pods).
+
+4. To find the IP address for your service, use
+   `kubectl get svc knative-ingressgateway -n istio-system` to get the ingress IP for your
+   cluster. If your cluster is new, it may take sometime for the service to get asssigned
+   an external IP address.
+
+    ```shell
+    kubectl get svc knative-ingressgateway -n istio-system
+
+    NAME                     TYPE           CLUSTER-IP     EXTERNAL-IP      PORT(S)                                      AGE
```

Comment:
The output of the command should be in a separate shell block so users can copy the entire command block without interpreting its contents.